tools/distpartitioning/data_shuffle.py

```diff
@@ -12,82 +12,88 @@
-    Prepare the list of slices targeting each of the process and
-    trigger alltoallv_cpu for the message exchange.
+def gen_node_data(rank, world_size, node_part_ids, ntid_cnt_map, ntid_ntype_map, nid_schema_map):
+    '''
+    For this data processing pipeline, reading node files is not needed. All the needed information about
+    the nodes can be found in the metadata json file. This function generates the nodes owned by a given
+    process, using metis partitions. 
 
-    Parameters:
+    Parameters: 
     -----------
     rank : int
-        rank of the current process
+        rank of the process
     world_size : int
-        total no. of participating processes
-    node_data : dictionary
-        nodes data dictionary with keys as column names and values as
-        columns from the nodes csv file
+        total no. of processes
+    node_part_ids : 
+        numpy array, whose length is same as no. of nodes in the graph. Index in this array is the global_nid
+        and value is the partition-id which owns the node
+    ntid_cnt_map : 
+        a dictionary where keys are node_type ids and values are total nodes which are of that node_type
+    ntid_ntype_map : 
+        a dictionary where keys are node_type ids and values are node_type names
+    nid_schema_map: 
+        a dictionary, which is extracted from the input graph metadata json file for node information, 
+        key is `nid`
 
     Returns:
     --------
     dictionary : 
-        the input argument, node_data, is updated with the node data received by other processes
-        in the world.
-    """
+        dictionary where keys are column names and values are numpy arrays, these arrays are generated by 
+        using information present in the metadata json file
+
+    '''
+    '''
+    ntypes_nid_map -> ntype_name -> str(ntype_id)
+    ntypes_id_count -> str(ntype_id) -> total node count
+    '''
+
+    local_node_data = {}
+    gnid_start = 0
+    gnid_end = 0
+    for ntypeid in range(len(ntid_cnt_map)):
+        ntype_name = ntid_ntype_map[str(ntypeid)]
+        ntype_info = nid_schema_map[ntype_name]
+        type_start = int(ntype_info["data"][0][1])
+        type_end = int(ntype_info["data"][-1][2])
+
+        gnid_end += type_end
+
+        node_partid_slice = node_part_ids[gnid_start:gnid_end]
+        cond = node_partid_slice == rank
+        own_gnids = np.arange(gnid_start, gnid_end, dtype=np.int64)
+        own_gnids = own_gnids[cond == 1]
+
+        own_tnids = np.arange(type_start, type_end, dtype=np.int64)
+        own_tnids = own_tnids[cond == 1]
+        if constants.NTYPE_ID in local_node_data:
+            local_node_data[constants.NTYPE_ID] = np.concatenate([local_node_data[constants.NTYPE_ID],
+                                                                    np.ones(own_gnids.shape, dtype=np.int64)*ntypeid])
+        else:
+            local_node_data[constants.NTYPE_ID] = np.ones(own_gnids.shape, dtype=np.int64)*ntypeid
```

Comment:
why don't you do `local_node_data[constants.NTYPE_ID].append(np.ones(own_gnids.shape, dtype=np.int64)*ntypeid]))` and concatenate everything in the end?

Reply:
In this way, the code is more concise and also more efficient

Reply:
Agreed. Changed the code accordingly. Now we concatenate in a single shot after all the data is collected.